contract/src/common/asserts.rs

```diff
@@ -13,3 +14,7 @@ impl Contract {
         require!(self.tokens.owner_by_id.get(token_id) == Some(owner_id.clone()), "Account doesnt own the token");
     }
 }
+
+pub(crate) fn assert_enough_gas(required: Gas) {
+    require!(remaining_gas() >= required, "Not enough gas for further operations");
```

Comment:
[nit]

Would be neat to also show in this error message how much gas was attached, how much is required and how much left.